private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -110,7 +110,11 @@ export function updateTemplate(templateID?: string, templateJSON?: string, sampl
 
     let newTemplate = new PostedTemplate();
     const id = templateID || appState.currentTemplate.templateID;
-
+    
+    const version = currentVersion || 
+    (appState.currentTemplate.template && appState.currentTemplate.template.instances && appState.currentTemplate.template.instances.length > 0 ?
+    appState.currentTemplate.template!.instances![0].version : "1.0")
```

Comment:
firstly, are the `!` needed in this context? 

secondly, what is the expected version from `appState.currentTemplate.template!.instances![0].version`? Is that the latest or the oldest available version?